docs/events.md

```diff
@@ -0,0 +1,671 @@
+  - [device.register](#device-register)
+  - [device.unregister](#device-unregister)
+  - [device.schema.sent](#device-schema-sent)
+  - [data.sent](#data-sent)
+  - [data.request](#data-request)
+  - [data.update](#data-update)
+  - [device.list](#device-list)
+  - [device.auth](#device-auth)
+
+- [Subscribe](#Subscribe) (external clients can subscribe to):
+  - [device.registered](#device-registered)
+  - [device.unregistered](#device-unregistered)
+  - [device.schema.updated](#device-schema-updated)
+  - [data.published](#data-published)
+  - [device.[id].data.request](#device-<id>-data-request)
+  - [device.[id].data.update](#device-<id>-data-update)
+
+-----------------------------------------------------------------
+
+## Publish
+
+This section describes the events that this service can receive from the external applications.
+
+### **device.register** <a name="device-register"></a>
+
+Event-command to register a new thing on the things registry. The operation response is sent through [`device.registered`](#device-registered) event.
+
+<details>
+  <summary>Headers</summary>
+
+  - `token` **String** user's token
+
+</details>
+
+<details>
+  <summary>Payload</summary>
+
+  JSON in the following format:
+
+  - `id` **String** thing's ID
+  - `name` **String** thing's name
+
+  Example:
+
+  ```json
+  {
+    "id": "fbe64efa6c7f717e",
+    "name": "KNoT Thing"
+  }
+  ```
+</details>
+
+<details>
+  <summary>AMQP Binding</summary>
+
+  - AMQP:
+    - Exchange:
+      - Type: direct
+      - Name: device
+      - Durable: `true`
+      - Auto-delete: `false`
+    - Routing key: device.register
+
+</details>
+
+### **device.unregister** <a name="device-unregister"></a>
+
+Event-command to remove a thing from the things registry. The operation response is sent through [`device.unregistered`](#device-registered) event.
+
+<details>
+  <summary>Headers</summary>
+
+  - `token` **String** user's token
+
+</details>
+
+<details>
+  <summary>Payload</summary>
+
+  JSON in the following format:
+
+  - `id` **String** thing's ID
+
+  Example:
+
+  ```json
+  {
+    "id": "fbe64efa6c7f717e"
+  }
+  ```
+</details>
+
+<details>
+  <summary>AMQP Binding</summary>
+
+  - Exchange:
+    - Type: direct
+    - Name: device
+    - Durable: `true`
+    - Auto-delete: `false`
+  - Routing key: device.unregister
+
+</details>
+
+### **device.list** <a name="device-list"></a>
+
+Event-command to list the registered things. It follows the request/reply pattern. After obtaining the things, `babeltower` will send a reply message by using the `correlation_id` property, which is received in the request header, as its `routing_key`. Because of that, considering the **requestor** has created and sent this `correlation_id` in the request, it can also subscribe to receive events that arrive with a `routing_key` equivalent to the `correlation_id`. Therefore, the reply is received by the application that has sent the request, in a **one-to-one** manner.
```

Comment:
Even better! :smile: 

A quick suggestion, tho! `its` is quite ambiguous as it may refer to `babeltower` or `reply message` :grimacing: 

```suggestion
Event-command to list the registered things. It follows the request/reply pattern. After obtaining the things, `babeltower` will send a reply message by using the `correlation_id` property, which was received in the request header, as reply message's  `routing_key`. Because of that, considering the **requestor** has created and sent this `correlation_id` in the request, it can also subscribe to receive events that arrive with a `routing_key` equivalent to the `correlation_id`. Therefore, the reply is received by the application that has sent the request, in a **one-to-one** manner.
```

Reply:
Done.